__tests__/data/zendesk_input.json

```diff
@@ -1,4 +1,65 @@
 [
+  {
+    "destination": {
+      "Config": {
+        "apiToken": "kFLaFcndBR9YHFxNVp7kQ5xxLr9syLWoE8jwot60",
+        "createUsersAsVerified": true,
+        "domain": "rudderlabshelp",
+        "email": "prabrisha@rudderlabs.com",
+        "password": "Ey!i@2tn6YhSDpx",
+        "removeUsersFromOrganization": true,
+        "sendGroupCallsWithoutUserId": true
+      },
+      "DestinationDefinition": {
+        "DisplayName": "Zendesk",
+        "ID": "1YknZ1ENqB8UurJQJE2VrEA61tr",
+        "Name": "ZENDESK"
+      },
+      "Enabled": true,
+      "ID": "1Z3zFXE6zwvNJBVOUzCuJxeO51P",
+      "Name": "zendesk",
+      "Transformations": []
+    },
+    "message": {
+      "anonymousId": "297b0750-934b-4411-b66c-9b418cdbc0c9",
+      "channel": "web",
+      "context": {
+        "app": {
+          "build": "1.0.0",
+          "name": "RudderLabs JavaScript SDK",
+          "namespace": "com.rudderlabs.javascript",
+          "version": "1.1.0-beta.2"
+        },
+        "ip": "0.0.0.0",
+        "library": {
+          "name": "RudderLabs JavaScript SDK",
+          "version": "1.1.0-beta.2"
+        },
+        "locale": "en-GB",
+        "os": {
+          "name": "",
+          "version": ""
+        },
+        "screen": {
+          "density": 2
+        },
+        "traits": {
+        },
```

Comment:
For track events, there is also an edge case where a missing `name` field in `traits` can cause an exception (while trying to create a new user). So, can we modify this test case input exactly to cover the missing `email` case?

Reply:
this test case is for the first error, where if none of the traits in identify mapping is present user is not created inside payload.
We can add test case to show the 2nd error as well.

Reply:
For the first error, just the absence of `email` is enough. I believe this data is good enough for the second error but might need more work to cover that.

Reply:
They both are of different eventType (track and identify) therefore included a new test case to demonstrate email misssing in traits error.

Reply:
I was referring to the exception thrown in `createUser` method.

Reply:
Sorry. `createUser` method is being in two different places. Inside `processTrack` and in `processGroup`. 

- For `processTrack` if we don't have `userEmail` we are throwing error so it won't reach `createUser` without `userEmail` . And we don't want to make call without a valid `userEmail` .
- For `processGroup` there is already a check present for `traits.name` and `traits.email`

Reply:
Agreed. In order to cover the first error, the input should just not contain `email` field and for the second one (in `createUser`), at least the `name` field should be missing in the `traits` object.
We should have test cases to explicitly cover each of them.